app/controllers/course/learning_map_controller.rb

```diff
@@ -87,29 +91,36 @@ def map_conditionals_to_nodes
     generate_node_depths(nodes)
   end
 
-  def generate_all_node_relations
+  def generate_all_node_relations # rubocop:disable Metrics/AbcSize
     relations = init_all_node_relations
     node_ids_to_children = relations[:node_ids_to_children]
     node_ids_to_parents = relations[:node_ids_to_parents]
+    node_ids_to_unlock_level = relations[:node_ids_to_unlock_level]
 
     @conditionals.each do |conditional|
       node_id = get_node_id(conditional)
 
       conditional.conditions.each do |condition|
-        next if condition.actable_type == Course::Condition::Level.name
+        if condition.actable_type == Course::Condition::Level.name
+          level_condition = Course::Condition::Level.find(condition.actable_id)
+          node_ids_to_unlock_level[node_id] = level_condition.minimum_level
+          next
```

Comment:
Perhaps can test with level conditions.

Reply:
Disabled lint since it's only 3 lines above the limit and I think creating another method adds an unnecessary layer of abstraction.

Level conditions tested on the frontend and they render correctly; will create a separate PR in future to include more comprehensive tests for all the different condition types.

Reply:
Roger, thanks for clarifying!